### DISCUSSION TITLE
Drag and Drop using react DND.

### DISCUSSION THREAD

**Discussion by ParikshitGit1** [cypress-io/cypress#30677] [Questions and Help]
I am trying to automate drag and drop function using cypress with unique selectors but after execution cypress runner is not able to locate the ID attribute even it is unique. Can anyone please provide any solution on this. Providing URL & code for reference.

https://stackblitz.com/edit/react-dnd-ve3iqo?file=Card.js,Container.js,index.html,index.js 

describe('sample', () => {

    it('fsga', () => {

        cy.visit('https://stackblitz.com/edit/react-dnd-ve3iqo?file=index.html,')

        cy.get('#sample1').
            drag('#demo11')

    })
})

**Comment by portyu9** [comment]
Hi @ParikshitGit1 

The selector issue is happening before drag-and-drop: `cy.visit()` is opening the **StackBlitz editor**, not the rendered application. `cy.get('#sample1')` only searches the current AUT document, so it will not find an element that lives inside StackBlitz's preview frame.

Use the application's standalone preview URL (or host the reproduction directly) and first verify both elements are in the AUT DOM:

```js
cy.visit('<standalone-preview-url>')

cy.get('#sample1').should('exist')
cy.get('#demo11').should('exist')
```

Also, `.drag()` is not a built-in Cypress command; it comes from a plugin. For React DnD using the HTML5 backend, the current Cypress plugin directory lists `cypress-real-dnd`, which performs real HTML5 drag/drop events and explicitly supports `react-dnd`.

So I would fix the target URL first. If the selectors then resolve but the drag still does not work, use an HTML5/CDP-based drag implementation rather than relying on synthetic drag events.

Please mark this answer as accepted if it helped, thank you.
